### PR TITLE
TE-4494 Added additional BC check for enabling mapping resources in t…

### PR DESCRIPTION
**Pull request by SArtemV**
- Developer(s): @asaulenko 

- Peer Reviewer:

- Ticket: https://spryker.atlassian.net/browse/TE-4494

- Release group: https://release.spryker.com/release-groups/view/2305

- Academy PR: ACADEMY_URL_HERE

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department
- [ ] All new or heavily changed classes get an "A" rating (Scrutinizer), except Facades, Factories, QueryContainers and Tests

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   SynchronizationBehavior  | patch          |                       |

#### Release Notes

This release updated the synchronization module to the latest version.

-----------------------------------------

#### Module SynchronizationBehavior

_**Patch:** Backwards-compatible bug fix_

Def of done (by responsible developer):
- [ ] All changes are backward-compatible. Outdated code is marked as deprecated.
- [ ] The change is isolated and not mixed with any cleanups or other changes.
- [ ] New and changed facade methods covered by functional tests. / Has nothing to test.
- [ ] New and changed complex business logic is covered by unit or integration tests. / Has nothing to test.

##### Change log

Improvements

- Added additional BC check for enabling mapping resources in the legacy environment

### CODE REVIEW

**Review by lockw1n — APPROVED** [review]

**Review by nuzzy — APPROVED** [review]

File: src/Spryker/Zed/SynchronizationBehavior/Persistence/Propel/Behavior/SynchronizationBehavior.php [hunk 1]
```diff
@@ -184,7 +184,10 @@ public function modifyTable()
             }
         }
 
-        if (isset($parameters['mapping_resource'])) {
+        /**
+         * @deprecated Will be removed without replacement.
```
Comment (gechetspr) [inline src/Spryker/Zed/SynchronizationBehavior/Persistence/Propel/Behavior/SynchronizationBehavior.php]: I think we don't need deprecation messages inside the method body. The check method is deprecated already

**Review by gechetspr — CHANGES_REQUESTED** [review]
Small thing to fix. Also please add a RG and BC check branch.

**Review by alexanderM91 — APPROVED** [review]

File: src/Spryker/Zed/SynchronizationBehavior/Persistence/Propel/Behavior/SynchronizationBehavior.php [hunk 2]
```diff
@@ -183,8 +183,8 @@ public function modifyTable()
                 ]);
             }
         }
-
-        if (isset($parameters['mapping_resource'])) {
+
```
Comment (dereuromark) [inline src/Spryker/Zed/SynchronizationBehavior/Persistence/Propel/Behavior/SynchronizationBehavior.php]: whitespace noise, please configure your IDE properly.

Comment (ZakirovDamir) [reply src/Spryker/Zed/SynchronizationBehavior/Persistence/Propel/Behavior/SynchronizationBehavior.php]: Removed

**Review by gechetspr — APPROVED** [review]

**Pull request merged**